chipsec: 1.10.6 -> 1.13.17

EclStas opened the pull request:
Update chipsec source to point to latest release tag, 1.13.17, switching to using `pyproject = true` per latest recommendations.

* Use chipsec marker file `README.NO_KERNEL_DRIVER` to signal when the driver shouldn't be built. I couldn't find a way to pass `--skip-driver` to the setup script.
* Redirect chipsec logging to "/tmp/chipsec/" since otherwise chipsec tries to dump logs into its installation directory (and into the nix store)
* Delete unused ".diff"s

## Things done

- Built on platform:
  - [X] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [X] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [X] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [X] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

EclStas commented:
I suspect we're going to run into this issue: https://github.com/chipsec/chipsec/issues/2563 
I'm hoping `nixpkgs-review` will be be able to help identify which kernel versions we might have to mark as "broken".

EclStas commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 454660`
Commit: `d06c32a0b2b45e28a58ec7ff3a0eab3faf7fdabb`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 24 packages built:</summary>
  <ul>
    <li>chipsec</li>
    <li>chipsec.dist</li>
    <li>linuxKernel.packages.linux_5_10.chipsec</li>
    <li>linuxKernel.packages.linux_5_10.chipsec.dist</li>
    <li>linuxKernel.packages.linux_5_15.chipsec</li>
    <li>linuxKernel.packages.linux_5_15.chipsec.dist</li>
    <li>linuxKernel.packages.linux_6_1.chipsec</li>
    <li>linuxKernel.packages.linux_6_1.chipsec.dist</li>
    <li>linuxPackages.chipsec (linuxKernel.packages.linux_6_12.chipsec)</li>
    <li>linuxPackages.chipsec.dist (linuxKernel.packages.linux_6_12.chipsec.dist)</li>
    <li>linuxKernel.packages.linux_hardened.chipsec (linuxKernel.packages.linux_6_12_hardened.chipsec)</li>
    <li>linuxKernel.packages.linux_hardened.chipsec.dist (linuxKernel.packages.linux_6_12_hardened.chipsec.dist)</li>
    <li>linuxPackages_latest.chipsec (linuxKernel.packages.linux_6_17.chipsec)</li>
    <li>linuxPackages_latest.chipsec.dist (linuxKernel.packages.linux_6_17.chipsec.dist)</li>
    <li>linuxKernel.packages.linux_6_6.chipsec</li>
    <li>linuxKernel.packages.linux_6_6.chipsec.dist</li>
    <li>linuxPackages_lqx.chipsec (linuxKernel.packages.linux_lqx.chipsec)</li>
    <li>linuxPackages_lqx.chipsec.dist (linuxKernel.packages.linux_lqx.chipsec.dist)</li>
    <li>linuxPackages_xanmod.chipsec (linuxKernel.packages.linux_xanmod.chipsec)</li>
    <li>linuxPackages_xanmod.chipsec.dist (linuxKernel.packages.linux_xanmod.chipsec.dist)</li>
    <li>linuxPackages_xanmod_latest.chipsec (linuxKernel.packages.linux_xanmod_latest.chipsec, linuxPackages_xanmod_stable.chipsec)</li>
    <li>linuxPackages_xanmod_latest.chipsec.dist (linuxKernel.packages.linux_xanmod_latest.chipsec.dist, linuxPackages_xanmod_stable.chipsec.dist)</li>
    <li>linuxPackages_zen.chipsec (linuxKernel.packages.linux_zen.chipsec)</li>
    <li>linuxPackages_zen.chipsec.dist (linuxKernel.packages.linux_zen.chipsec.dist)</li>
  </ul>
</details>

EclStas commented:
Hi @erdnaxe  @JohnAZoidberg ! Could you guys please help review my PR? I'm new to nix and appreciate any feedback :) 

Thanks in advance

eljamm commented on pkgs/tools/security/chipsec/default.nix:
```diff
@@ -43,36 +35,28 @@ python3.pkgs.buildPythonApplication rec {
-  nativeCheckInputs = with python3.pkgs; [
-    distro
-    pytestCheckHook
+  propagatedBuildInputs = with python3.pkgs; [
```
```suggestion
  dependencies = with python3.pkgs; [
```

eljamm commented on pkgs/tools/security/chipsec/default.nix:
```diff
@@ -11,28 +11,20 @@
+  version = "1.13.17";
+
+  pyproject = true;
+  build-system = [ python3.pkgs.setuptools ];
```
It's probably be better to move this near `dependencies`, so all the inputs are grouped together.

EclStas commented on pkgs/tools/security/chipsec/default.nix:
```diff
@@ -11,28 +11,20 @@
+  pyproject = true;
+  build-system = [ python3.pkgs.setuptools ];
 
   disabled = !stdenv.hostPlatform.isLinux;
```
Great point, thanks for catching that. I'm going to try removing this and adding some more platforms where I believe `chipsec` should work with the "no driver" mode.

EclStas commented:
Hi @eljamm , thank you for taking the time to review!
 
I'd like to see if `chipsec` will work with more platforms: it can be pretty useful even without the driver (e.g. to unpack uefi FV volumes). So I guess we have to wait for CI/`nixpkgs-review` to tell us if I have to restrict the platforms.

eljamm commented:
I use [nixpkgs-review-gha](https://github.com/Defelo/nixpkgs-review-gha) to run `nixpkgs-review` on platforms that are unavailable to me, and the [build I ran](https://github.com/eljamm/nixpkgs-review-gha/actions/runs/19374104510/job/55437092655) timed out on `x86_64-linux` because it had to compile a lot of things, so I ran it on my machine, instead. This said, the `chipsec` executable seems to be building successfully on [aarch64-darwin](https://github.com/eljamm/nixpkgs-review-gha/actions/runs/19374104510/job/55437092655#step:9:30), [x86_64-darwin](https://github.com/eljamm/nixpkgs-review-gha/actions/runs/19374104510/job/55437092693#step:9:30) and [aarch64-linux](https://github.com/eljamm/nixpkgs-review-gha/actions/runs/19374104510/job/55437092660#step:9:31).

This looks really good, @EclStas. If you'd just squash all the commits, as per [the commit conventions](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions), I think this would be ready to be merged.

EclStas committed:
chipsec: 1.10.6 -> 1.13.17

Update chipsec source to point to latest release tag,
switching to using `pyproject = true` per latest recommendations.

* Use chipsec marker file `README.NO_KERNEL_DRIVER` to signal when the
  driver shouldn't be built. I couldn't find a way to pass
`--skip-driver` to the setup script.
* Redirect chipsec logging to "/tmp/chipsec/" since otherwise
  chipsec tries to dump logs into its installation directory
  (into the nix store)
* Add additional supported platforms (without kernel driver)
  * `aarch64-darwin`
  * `x86_64-darwin`
  * `aarch64-linux`
* Install the kernel module so it can be manually loaded
* Set `chipsec_main` as the main program for chipsec

eljamm reviewed (commented):
> Thank you for running the builds and explaining your process @eljamm : I learned a lot from this!

Absolutely. You've done a great job unbreaking this, already, but I'm glad you found my comments useful.

EclStas commented on pkgs/tools/security/chipsec/default.nix:
Ah I think I see what's going on! `pkgs.chipsec` always builds with the option `withDriver = false`.
If I do this things seem to work well:

```nix
{
  name = "chipsec";

  nodes.machine =
    { pkgs, config, ... }:
    {
      environment.systemPackages = [
        config.boot.kernelPackages.chipsec
      ];
    };

  testScript = ''
    start_all()
    machine.wait_for_unit("multi-user.target")
    machine.succeed("chipsec_main --help")
    machine.succeed("chipsec_util spi info")
  '';
}
```
I added `spi info` to the test since that one certainly requires the kernel driver.

eljamm replied:
Nice. I think supporting both loading methods would probably be the way to go. I'm not familiar with how chipsec is used, but it might be safe to assume that some people might not want to use the CLI each time to load the module.

EclStas commented:
@eljamm I think we resolved all the outstanding items. Anything else we should do before this is ready to merge?

eljamm commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 454660`
Commit: [`7b85e46e822b057fdc0a368cd61713a7fa57314f`](https://github.com/NixOS/nixpkgs/commit/7b85e46e822b057fdc0a368cd61713a7fa57314f) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/7b85e46e822b057fdc0a368cd61713a7fa57314f..pull/454660/head))
Merge: [`b06b6b04299ca60ca674f6b6a40a918c1ea06d5b`](https://github.com/NixOS/nixpkgs/commit/b06b6b04299ca60ca674f6b6a40a918c1ea06d5b)

Logs: https://github.com/eljamm/nixpkgs-review-gha/actions/runs/19661889388


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 24 packages built:</summary>
  <ul>
    <li>chipsec</li>
    <li>chipsec.dist</li>
    <li>linuxKernel.packages.linux_5_10.chipsec</li>
    <li>linuxKernel.packages.linux_5_10.chipsec.dist</li>
    <li>linuxKernel.packages.linux_5_15.chipsec</li>
    <li>linuxKernel.packages.linux_5_15.chipsec.dist</li>
    <li>linuxKernel.packages.linux_6_1.chipsec</li>
    <li>linuxKernel.packages.linux_6_1.chipsec.dist</li>
    <li>linuxPackages.chipsec (linuxKernel.packages.linux_6_12.chipsec)</li>
    <li>linuxPackages.chipsec.dist (linuxKernel.packages.linux_6_12.chipsec.dist)</li>
    <li>linuxKernel.packages.linux_hardened.chipsec (linuxKernel.packages.linux_6_12_hardened.chipsec)</li>
    <li>linuxKernel.packages.linux_hardened.chipsec.dist (linuxKernel.packages.linux_6_12_hardened.chipsec.dist)</li>
    <li>linuxPackages_latest.chipsec (linuxKernel.packages.linux_6_17.chipsec)</li>
    <li>linuxPackages_latest.chipsec.dist (linuxKernel.packages.linux_6_17.chipsec.dist)</li>
    <li>linuxKernel.packages.linux_6_6.chipsec</li>
    <li>linuxKernel.packages.linux_6_6.chipsec.dist</li>
    <li>linuxPackages_lqx.chipsec (linuxKernel.packages.linux_lqx.chipsec)</li>
    <li>linuxPackages_lqx.chipsec.dist (linuxKernel.packages.linux_lqx.chipsec.dist)</li>
    <li>linuxPackages_xanmod.chipsec (linuxKernel.packages.linux_xanmod.chipsec)</li>
    <li>linuxPackages_xanmod.chipsec.dist (linuxKernel.packages.linux_xanmod.chipsec.dist)</li>
    <li>linuxPackages_xanmod_latest.chipsec (linuxKernel.packages.linux_xanmod_latest.chipsec, linuxPackages_xanmod_stable.chipsec)</li>
    <li>linuxPackages_xanmod_latest.chipsec.dist (linuxKernel.packages.linux_xanmod_latest.chipsec.dist, linuxPackages_xanmod_stable.chipsec.dist)</li>
    <li>linuxPackages_zen.chipsec (linuxKernel.packages.linux_zen.chipsec)</li>
    <li>linuxPackages_zen.chipsec.dist (linuxKernel.packages.linux_zen.chipsec.dist)</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>chipsec</li>
    <li>chipsec.dist</li>
  </ul>
</details>

---
### `x86_64-darwin` (sandbox = relaxed)
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>chipsec</li>
    <li>chipsec.dist</li>
  </ul>
</details>

---
### `aarch64-darwin` (sandbox = relaxed)
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>chipsec</li>
    <li>chipsec.dist</li>
  </ul>
</details>

eljamm reviewed (approved):
> @eljamm I think we resolved all the outstanding items. Anything else we should do before this is ready to merge?

This looks good to me. Now we only need to wait for someone to merge it.

Hydra: https://hydra.nixos.org/eval/1820385?filter=chipsec&compare=1820337&full=
ZHF: https://github.com/NixOS/nixpkgs/issues/457852

eljamm commented:
Actually, what about including the [NixOS test](https://github.com/NixOS/nixpkgs/pull/454660#discussion_r2547925550) as well? (can be done in a followup)

risicle reviewed (approved):
`nixpkgs-review` happy, macos 14 x86_64, nixos x86_64 & aarch64-linux

risicle commented on pkgs/tools/security/chipsec/log-path.diff:
```diff
@@ -0,0 +1,22 @@
+         self.logfile = None
+         self.ALWAYS_FLUSH = False
+-        self.LOG_PATH = os.path.join(BASE_PATH, "logs")
++        self.LOG_PATH = "/tmp/chipsec"
```
Come to think this might not be a great place to send logs if they may contain security-sensitive information - it's not known what the ownership of this path is on any system.
